specs/phase0/p2p-interface.md

```diff
@@ -219,7 +219,13 @@ The following gossipsub [parameters](https://github.com/libp2p/specs/tree/master
 
 ### Topics and messages
 
-Topics are plain UTF-8 strings and are encoded on the wire as determined by protobuf (gossipsub messages are enveloped in protobuf messages). Topic strings have form: `/eth2/TopicName/TopicEncoding`. This defines both the type of data being sent on the topic and how the data field of the message is encoded.
+Topics are plain UTF-8 strings and are encoded on the wire as determined by protobuf (gossipsub messages are enveloped in protobuf messages). Topic strings have form: `/eth2/ForkDigest/Name/Encoding`. This defines both the type of data being sent on the topic and how the data field of the message is encoded.
+
+- `ForkDigest` - the lowercase hex-encoded (no "0x" prefix) bytes of `ForkDigest(compute_fork_data_root(current_fork_version, genesis_validators_root)[:4])` where
+    - `current_fork_version` is the fork version of the epoch of the message to be sent on the topic
```

Comment:
What about slashings though? they span 2 epochs, potentially. We need to send to both, or have everyone stay on multiple slashing topics for as long as slashing is possible.

We can also leave it unspecified, noting that only one fork version exists for now, and that the hard fork will decide the upgrade procedure.

Reply:
I asked about what will happen if slashing messages formats are different cross forks: https://github.com/ethereum/eth2.0-specs/issues/1475
It seems the status-quo is to believe in the honest majority.

Reply:
We can make a table of messages and related field that dictates the epoch/gossipchannel.

As for cross-fork slashings, it is a more fundamental problem than just which channel to send it on. To catch slashings across a fork boundary in which the data-type changed would require specific data structures to handle mismatched types and separate functions for verifying validity of each. As discussed in #1475, I'm happy to sweep that under the rug in early phases and address it if/when we need to.

When we upgrade to Phase 1, we'll make it clear that fraud proofs across the boundary are not accepted and that this channel upgrade path only supports message types from the respective fork